spring-cloud-azure-feature-management-web/src/main/java/com/microsoft/azure/spring/cloud/feature/manager/FeatureHandler.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.spring.cloud.feature.manager;
+
+import java.io.IOException;
+import java.lang.reflect.Method;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.web.method.HandlerMethod;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+
+/**
+ * Intercepter for Requests to check if they should be run.
+ */
+@Component
+public class FeatureHandler extends HandlerInterceptorAdapter {
+
+    Logger logger = LoggerFactory.getLogger(FeatureHandler.class);
+
+    @Autowired
+    FeatureManager featureManager;
+
+    @Autowired
+    FeatureManagerSnapshot featureManagerSnapshot;
```

Comment:
About the `@Autowired`, in general, it's more recommended to use constructor injection instead of `@Autowired` injection, considering from test perspective. But may change this later when testing.